plugin/src/main/groovy/com/novoda/staticanalysis/internal/ktlint/KtlintConfigurator.groovy

```diff
@@ -84,12 +84,19 @@ class KtlintConfigurator implements Configurator {
     }
 
     private void configureAndroidWithVariants(def mainVariants) {
-        mainVariants.all { configureKtlint(it.name) }
-        variantFilter.filteredTestVariants.all { configureKtlint(it.name) }
-        variantFilter.filteredUnitTestVariants.all { configureKtlint(it.name) }
+        mainVariants.all { configureAndroidVariant(it) }
+        variantFilter.filteredTestVariants.all { configureAndroidVariant(it) }
+        variantFilter.filteredUnitTestVariants.all { configureAndroidVariant(it) }
     }
 
-    private void configureKtlint(def sourceSetName) {
+    private void configureAndroidVariant(def variant) {
+        variant.sourceSets.each { sourceSet ->
+            configureKtlint(sourceSet.name)
+        }
+        configureKtlint(variant.name)
```

Comment:
isn't this already covered by L93-L95? also, is it possible to use `.all {}` here instead?